File: dichotomic_search/dichotomic_search.js
function dichotomic_search(array = [],v = 0) {

	let med = Math.round(array.length / 2);
	
	return 	array.length > 1 ? 
				array[med] > v ? 
					dichotomic_search(array.splice(0,med),v) : 
					dichotomic_search(array.splice(med,array.length),v) : 

				array[0] == v ? 
					v : 
					null;
}

let result = dichotomic_search();

File: 4_dichotomic_search/dichotomic_search.py
# coding: utf8
# Python implementation of Dichotomic Search algo
# Author: Eric Alcaide

def dicho(arr = [], v=0):
	med = round(len(arr)/2)
	if len(arr) <= 1:
		return True if arr[0] == v else False
	else: 
		return dicho(arr[:med], v) if arr[med] > v else dicho(arr[med:], v)


result = dicho(['a', 'b', 'c'], 'c')
print(result)
